cpp/test/handle.cpp

```diff
@@ -24,65 +24,41 @@ namespace raft {
 
 TEST(Raft, HandleDefault) {
   handle_t h;
-  ASSERT_EQ(0, h.get_num_internal_streams());
+  ASSERT_EQ(0, h.get_stream_pool().get_pool_size());
   ASSERT_EQ(0, h.get_device());
-  ASSERT_EQ(nullptr, h.get_stream());
+  ASSERT_EQ(rmm::cuda_stream_default, h.get_stream());
   ASSERT_NE(nullptr, h.get_cublas_handle());
   ASSERT_NE(nullptr, h.get_cusolver_dn_handle());
   ASSERT_NE(nullptr, h.get_cusolver_sp_handle());
   ASSERT_NE(nullptr, h.get_cusparse_handle());
 }
 
 TEST(Raft, Handle) {
-  handle_t h(4);
-  ASSERT_EQ(4, h.get_num_internal_streams());
+  // test stream pool creation
+  rmm::cuda_stream_pool stream_pool{4};
```

Comment:
To avoid a "magic" number in the assertion below, could we declare the pool size as a variable and reuse it as we do with `n_streams` at line 53?

Reply:
Done, thanks!